src/utils/work/fetchWork.js

```diff
@@ -139,30 +147,36 @@ const fetchWork = async (
   // Then, we may be able to find this in S3.
   let response = await seekFromS3(ETag, experimentId);
 
-  // If response cannot be fetched, go to the worker.
+  // If there is no response in S3, dispatch workRequest via the worker
+  let workResultReady = false;
   if (!response) {
-    response = await seekFromAPI(
+    workResultReady = await dispatchWorkRequest(
       experimentId,
       body,
       timeout,
       ETag,
-      eventCallback,
       {
         PipelineRunETag: qcPipelineStartDate,
         ...extras,
       },
     );
   }
 
-  if (!response) {
+  if (!workResultReady) {
```

Comment:
`workResultReady` would be false when the task timed out?

Reply:
No if it times out, it will throw an error. `workResultReady` is not needed, because it either returns true, or it throws an error.

Reply:
Could you check that the timeout works? Because locally I tried killing the worker and the timeout isn't reaching the handler to show the "try again"